sdk/middleware/sqecho/echo.go

```diff
@@ -34,8 +34,14 @@ import (
 func Middleware() echo.MiddlewareFunc {
 	return func(next echo.HandlerFunc) echo.HandlerFunc {
 		return func(c echo.Context) error {
-			// Create a new request record for this request.
 			req := c.Request()
+
+			if action := sdk.SecurityAction(req); action != nil {
+				action.Apply(c.Response())
+				return nil
```

Comment:
I assume this prevents other middlewares from running right?

Reply:
Yes:
1. You apply the security action the agent ask us to apply.
2. You abort, which is different depending on the frameworks used. Here, with Echo, it's returning nil.